src/app/multi-editor/multi-editor.component.ts

```diff
@@ -264,6 +265,17 @@ export class MultiEditorComponent implements OnInit {
     this.changeDetectorRef.markForCheck();
   }
 
+  checkExistenceOfAction(userActions: UserActions): boolean {
+    let existingAction = false;
+    userActions.actions.forEach(action => {
```

Comment:
use `some` for this, it's like old style `for` with `break;` when the check condition is true for the first time.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some